packages/node/src/lib/retry.ts

```diff
@@ -0,0 +1,94 @@
+import { AxiosError, AxiosInstance } from 'axios';
+import axiosRetry, { isNetworkError } from 'axios-retry';
+import { v4 as uuid } from 'uuid';
+import { INovuConfiguration } from './novu.interface';
+
+const NON_IDEMPOTENT_METHODS = ['post', 'patch'];
+const IDEMPOTENCY_KEY = 'Idempotency-Key'; // header key
+
+const DEFAULT_RETRY_MAX = 0;
+const DEFAULT_WAIT_MIN = 1;
+const DEFAULT_WAIT_MAX = 30;
+
```

Comment:
new global variables - for default values as suggested + I decided to move string value for header key to `IDEMPOTENCY_KEY` variable (to avoid "magic strings")